### DISCUSSION TITLE
[Esame 20 Luglio 2021] Esercizio 1

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/calcolo-delle-probabilita#259] [Esami - Bertini @ Matematica]
![Screenshot 2023-11-14 113005](https://github.com/sapienzastudentsnetwork/calcprob2324/assets/107944476/8188cd11-db31-4ac2-ace3-6d643d22e428)

**Comment by gpelisset** [comment]
Sia $A$ la variabile aleatoria che indica il numero della carta estratta da Alberto e $B_1$ la variabile aleatoria che indica il numero della prima carta estratta da Barbara e  $B_2$ la variabile aleatoria che indica il numero della seconda carta estratta da Barbara.
a) $P(A = 7) = 4 / 40 = 1 / 10$
    $$P((B_1, B_2) \in \\{(3, 4), (4, 3)\\}) = P((B_1, B_2) \in \\{(3, 4), (4, 3)\\}, A \in \\{3, 4\\}) + P((B_1, B_2) \in \\{(3, 4), (4, 3)\\}, A \notin \\{3, 4\\}) = (8 * 4 * 3) / (40 * 39 * 19) + (32 * 4 * 4) / (40 * 39 * 19) = 0.020512821$$
    dove ho disgregato l'evento e usato del semplice calcolo combinatorio
b) $P(A = 7, (B_1, B_2) \in \\{(3, 4), (4, 3)\\}) = (4 * 4 * 4) / (40 * 39 * 19) = 0.002159244$
c) $$P(B_1 + B_2 = 7|A \ge 8) = {\sum_{i = 1}}^{6} P(B_1 + B_2 = 7, B_1 = i|A \ge 8) = {\sum_{i = 1}}^6 P(B_2 = 7 - i, B_1 = i|A \ge 8) = {\sum_{i = 1}}^6 \dfrac{4 * 4}{39 * 38} = \dfrac{6 * 4 * 4}{39 * 38} = 0.064777328$$
d) $$P(A \ge 8|B_1 + B_2 = 7) = P(B_1 + B_2 = 7 | A \ge 8) P(A \ge 8) / P(B_1 + B_2 = 7) = 0.064777328 * (3 / 10) / (6 * 4  *4 / (40 * 39)) = 0.315789475$$

**Comment by gpelisset** [comment]
La bonta' dei risultati e' stata provata mediante la seguente simulazione
```python
import random


def test(iterations, *tests):
    print(", ".join(t.__name__ + ": " + str(sum(t() for _ in range(iterations)) / iterations) for t in tests))

class Card:
    __slot__ = (
            "number",
            "seed"
    )

    seeds = [*range(4)]

    def __init__(self, number, seed):
        self.number = number
        self.seed = seed

    @property
    def n(self): return self.number

cards = [Card(n, s) for s in Card.seeds for n in range(1, 11)]

# a)
def a1():
    random.shuffle(cards)
    return cards[0].n == 7

def a2():
    random.shuffle(cards)
    return {cards[0].n, cards[1].n} == {3, 4}

# b
def b():
    random.shuffle(cards)
    return (cards[0].n, {cards[1].n, cards[2].n}) == (7, {3, 4})

# c)
def c():
    random.shuffle(cards)
    while cards[0].n < 8:
        random.shuffle(cards)
    return cards[1].n + cards[2].n == 7

# d)
def d():
    random.shuffle(cards)
    while cards[0].n + cards[1].n != 7:
        random.shuffle(cards)
    return cards[2].n >= 8

test(100000, a1, a2, b, c, d)

```

**Comment by Elia-Belli** [comment]
**(1)** L'estrazione di $A$ è banale, per $B$ invece dobbiamo considerare anche l'ordine delle estrazioni:
$$\mathbb{P}(A=7)=\frac{4}{40}=\frac{1}{10}$$

$$\mathbb{P}((B_1=3,B_2=4) \cup (B_1=4,B_2=3))=2\cdot (\frac{4}{40}\cdot \frac{4}{39})$$

**(2)** 
$$\mathbb{P}((A=7)\cup (B\ estrae\ 3\ e\ 4))=\frac{1}{10}\cdot 2\cdot \frac{4}{39}\cdot \frac{4}{38}$$

**(3)** Il fatto che $A$ abbia rimosso una carta $\geq 8$ non influisce sul numero di modi in cui $B$ può ottenere $7$:
i modi per ottenere $7$ sono $6$ ovvero le mani $(1,6)(2,5)(3,4)(4,3)(5,2)(6,1)$ ognuna con probabilità $(\frac{4}{39}\cdot \frac{4}{38})$ di essere ottenuta
$$\mathbb{P}(B_1+B_2=7|A\geq 8)=6\cdot \frac{4}{39}\cdot \frac{4}{38}\simeq 0.06478$$

**(4)** Utilizzo la probabilità condizionata e Bayes:
$$\mathbb{P}(A\geq 8|B_1+B_2=7)=\frac{\mathbb{P}((A\geq 8) \cap (B_1+B_2=7))}{\mathbb{P}(B_1+B_2=7)}=\frac{\mathbb{P}(B_1+B_2=7| A\geq 8)\cdot \mathbb{P}(A\geq 8)}{\mathbb{P}(B_1+B_2=7)}$$

A questo punto sappiamo che $\mathbb{P}(A\geq 8)=4\cdot \frac{3}{40}=\frac{3}{10}$ e dobbiamo calcolare $\mathbb{P}(B_1+B_2=7)$ che è implicitamente condizionata dalla carta estratta da $A$, consideriamo i casi in cui l'estrazione non riduce i modi ottenere $7$ come nel punto **(3)** e quando influisce:

$$\mathbb{P}(B_1+B_2=7)=\mathbb{P}(B_1+B_2=7|A\geq 7)\cdot \mathbb{P}(A\geq 7)+\mathbb{P}(B_1+B_2=7|A\lt 7)\cdot \mathbb{P}(A\lt 7)=$$

Se $A$ pesca una delle carte che influiscono sulla somma, ovvero le carte $\lt 7$, avremo che la probabilità di due coppie (quelle che contegono la carta) diminuisce:

$$=(6\cdot \frac{4}{39}\cdot \frac{4}{38})\cdot \frac{16}{40}+[2\cdot \frac{3}{39}\cdot \frac{4}{38}+4\cdot \frac{4}{39}\cdot \frac{4}{38}]\cdot \frac{24}{40}\simeq 0.06152$$

Quindi svolgendo i calcoli si ottiene:

$$\mathbb{P}(A\geq 8|B_1+B_2=7)=0.31588$$

> E' possibile eseguire semplificazioni per ridurre i calcoli
